### PR TITLE
refactor: replace lodash & assertive usage

### PR DESCRIPTION
**Pull request by aaarichter**
- [x] refactor: remove lodash usage
- [x] refactor: use node assert
- [x] test: rewrite test with native assert 
- [x] chore: update packages

### CODE REVIEW

File: lib/browser/_asserters.js [hunk 1]
```diff
@@ -105,19 +105,22 @@ exports.fuzzyString = function assertFuzzyString(
...
   }
+
   const doc = `${selector} should ${shouldMatch ? '' : 'not '}have ${propName}`;
 
   return new Asserter(target => {
     return Promise.resolve(target)
-      .then(extract)
+      .then(res => {
+        return extract(res);
+      })
```
Comment (dbushong) [inline lib/browser/_asserters.js:121]: out of curiosity - why this change?

File: lib/browser/cookie.js [hunk 2]
```diff
@@ -66,19 +63,24 @@ exports.setCookies = util.deprecate(function setCookies(cookies) {
 
 exports.setCookieValues = function setCookieValues(cookies) {
   const browser = this;
+
   return Promise.all(
-    _.map(cookies, (value, name) => {
-      return browser.setCookieValue(name, value);
+    Object.entries(cookies).map(pair => {
+      return browser.setCookieValue(...pair);
```
Comment (dbushong) [inline lib/browser/cookie.js:69]: minor nit: while this is clever, if you ever end up wanting to type check this code, it'll work better to have something more explicit like:
```suggestion
    Object.entries(cookies).map(([k, v]) => {
      return browser.setCookieValue(k, v);
```

File: lib/browser/lighthouse.js [hunk 3]
```diff
@@ -33,8 +33,9 @@
 'use strict';
 
 const assert = require('assert');
-const isEmpty = require('lodash/isEmpty');
-const flatMap = require('lodash/flatMap');
+const flatMap = require('array.prototype.flatmap');
+
+flatMap.shim(); // TODO: remove with node 12
```
Comment (dbushong) [inline lib/browser/lighthouse.js:38]: I would recommend using it directly as a ponyfill - it's safer and not that much harder to refactor later.

File: lib/testium-driver-wd.js [hunk 4]
```diff
@@ -194,19 +201,19 @@ function initDriver(testium) {
...
-    if (!lighthouse) {
-      throw new Error('Lighthouse requires a newer version of node');
+  wd.addPromiseChainMethod(
+    'getLighthouseData',
+    function getLighthouseData(flags, config) {
+      if (!lighthouse) {
+        throw new Error('Lighthouse requires a newer version of node');
+      }
+      const devtoolsPort = testium.getChromeDevtoolsPort();
+      const options = Object.assign({ port: devtoolsPort }, flags);
```
Comment (dbushong) [inline lib/testium-driver-wd.js:211]:
```suggestion
      const options = { port: devtoolsPort , ...flags };
```

File: test/integration/dialog.test.js [hunk 5]
```diff
@@ -27,60 +27,102 @@ describe('dialogs', () => {
   describe('alert', () => {
     beforeEach(() => browser.clickOn('.link_to_open_an_alert'));
 
-    it('can get an alert text', async () => {
-      const text = await browser.getAlertText();
-      await browser.acceptAlert();
-      assert.equal('Alert text was not found', 'An alert!', text);
+    it('can get an alert text', done => {
```
Comment (dbushong) [inline test/integration/dialog.test.js:30]: Why'd these all change back to testing callback mode?

**Review by dbushong — APPROVED** [review]
Except for the polyfill and the callback test conversion commented on, LGTM!

File: lib/browser/_asserters.js (hunk 1 above)
Comment (aaarichter) [reply lib/browser/_asserters.js:121]: 💅  was testing something there and didn't revert

File: test/integration/dialog.test.js (hunk 5 above)
Comment (aaarichter) [reply test/integration/dialog.test.js:30]: I discovered that `acceptAlert()` and `dismissAlert()` are not part of the promise chain, but use WD commands directly.
Locally those tests run fine with `aync/await` but they are somewhat flaky in Travis

I'll promisify the them in a follow up.

**Comment by aaarichter** [comment]
Thanks for the review @dbushong . I addressed your feedback, but kept the prompt test changes for now. As mentioned, in the follow up, I'll promisify them, and revert the tests to the previous async await tests
